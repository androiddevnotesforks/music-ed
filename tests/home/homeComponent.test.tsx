import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom"
import HomeComponent from "../../components/home/homeComponent";

describe("HomeComponent", () => {
  beforeEach(() => {
    render(<HomeComponent />)
  })
  it("Renders", () => {
    const homeComponent = screen.getByTestId("home-component")
    expect(homeComponent).toBeInTheDocument()
  })
  it("Matches snapshot", () => {
    const homeComponent = screen.getByTestId("home-component")
    expect(homeComponent).toMatchSnapshot()
  })
  it("Home blurb is in the document", () => {
    const homeBlurb = screen.getByTestId("home-blurb")
    expect(homeBlurb).toBeInTheDocument()
  })
  it("Piano component is in the document", () => {
    const piano = screen.getByTestId("piano-div")
    expect(piano).toBeInTheDocument()
  })
  it("About link is in the document", () => {
    const aboutLink = screen.getByTestId("about-link")
    expect(aboutLink).toBeInTheDocument()
  })
  it("Exercises link is in the document", () => {
    const exercisesLink = screen.getByTestId("exercises-link")
    expect(exercisesLink).toBeInTheDocument()
  })

})