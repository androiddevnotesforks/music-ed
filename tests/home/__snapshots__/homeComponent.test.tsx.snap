// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeComponent Matches snapshot 1`] = `
<div
  class="flex flex-col items-center w-screen "
  data-testid="home-component"
>
  <div
    class="home-blurb"
    data-testid="home-blurb"
  >
    <h1
      class="p-2"
    >
      Welcome to the Aural Gymnasium.
    </h1>
    <p>
      This is a completely free resource for musicians to develop their relative pitch and chord recognition.
    </p>
    <p>
      You can use the exercises as much as you like. To track your progress, create an account.
    </p>
  </div>
  <div
    class="piano"
    data-testid="piano-div"
  >
    <button
      class="piano-btn"
      data-testid="note-name-toggle"
    >
      ABC
    </button>
    <div
      class="piano-keyboard"
    >
      <button
        class="text-sm font-thin white-key active:bg-red-400"
      />
      <button
        class="black-key text-sm font-thin active:bg-red-400"
      />
      <button
        class="text-sm font-thin white-key active:bg-red-400"
      />
      <button
        class="black-key text-sm font-thin active:bg-red-400"
      />
      <button
        class="text-sm font-thin white-key active:bg-red-400"
      />
      <button
        class="text-sm font-thin white-key active:bg-red-400"
      />
      <button
        class="black-key text-sm font-thin active:bg-red-400"
      />
      <button
        class="text-sm font-thin white-key active:bg-red-400"
      />
      <button
        class="black-key text-sm font-thin active:bg-red-400"
      />
      <button
        class="text-sm font-thin white-key active:bg-red-400"
      />
      <button
        class="black-key text-sm font-thin active:bg-red-400"
      />
      <button
        class="text-sm font-thin white-key active:bg-red-400"
      />
      <button
        class="text-sm font-thin white-key active:bg-red-400"
      />
      <button
        class="black-key text-sm font-thin active:bg-red-400"
      />
      <button
        class="text-sm font-thin white-key active:bg-red-400"
      />
      <button
        class="black-key text-sm font-thin active:bg-red-400"
      />
      <button
        class="text-sm font-thin white-key active:bg-red-400"
      />
      <button
        class="text-sm font-thin white-key active:bg-red-400"
      />
      <button
        class="black-key text-sm font-thin active:bg-red-400"
      />
      <button
        class="text-sm font-thin white-key active:bg-red-400"
      />
      <button
        class="black-key text-sm font-thin active:bg-red-400"
      />
      <button
        class="text-sm font-thin white-key active:bg-red-400"
      />
      <button
        class="black-key text-sm font-thin active:bg-red-400"
      />
      <button
        class="text-sm font-thin white-key active:bg-red-400"
      />
      <button
        class="text-sm font-thin white-key active:bg-red-400"
      />
    </div>
  </div>
  <div
    class="home-blurb"
  >
    <p>
      The Aural Gymnasium has over 40 exercises for musicians and students of any level to improve their aural skills. Answers are entered using the interactive keyboard.
    </p>
    <div
      class="flex flex-col items-center w-full "
    >
      <a
        href="/about"
      >
        <button
          class="text-green-500 border-green-500 hover:bg-green-50 active:bg-white rounded border p-4 m-2 font-extrabold text-center hover:bg-blue-50 active:bg-blue-100 hover:cursor-pointer"
          data-testid="about-link"
        >
          About
        </button>
      </a>
      <a
        href="/exercises"
      >
        <button
          class=" text-blue-500 border-blue-500 hover:bg-blue-50 active:bg-white rounded border p-4 m-2 font-extrabold text-center hover:bg-blue-50 active:bg-blue-100 hover:cursor-pointer"
          data-testid="exercises-link"
        >
          Get Started
        </button>
      </a>
    </div>
  </div>
</div>
`;
